src/inc/til/bitmap.h

```diff
@@ -382,6 +389,59 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         }
 
     private:
+        void translate_y(ptrdiff_t delta_y, bool fill)
+        {
+            if (delta_y == 0)
+            {
+                return;
+            }
+
+            const auto bitShift = delta_y * _sz.width();
+
+#pragma warning(push)
+            // we can't depend on GSL here (some libraries use BLOCK_GSL), so we use static_cast for explicit narrowing
+#pragma warning(disable : 26472)
+            const auto newBits = static_cast<size_t>(std::abs(bitShift));
+#pragma warning(pop)
+            const bool isLeftShift = bitShift > 0;
+
+            if (newBits >= _bits.size())
+            {
+                if (fill)
+                {
+                    set_all();
+                }
+                else
+                {
+                    reset_all();
+                }
+                return;
+            }
+
+            if (isLeftShift)
+            {
+                // This operator doesn't modify the size of `_bits`: the
+                // new bits are set to 0.
+                _bits <<= newBits;
+            }
+            else
+            {
+                _bits >>= newBits;
+            }
+
+            if (fill)
+            {
+                if (isLeftShift)
+                {
+                    _bits.set(0, newBits, true);
+                }
+                else
+                {
+                    _bits.set(_bits.size() - newBits, newBits, true);

```

Comment:
It's doing my head in. I can't figure out why _left shift_ is for a positive translation (downward)... but the tests seem to agree that this is correct, so ..?

Reply:
I understand, I was confused too at first 😅.

The library for `dynamic_bitset` stores the bits in an unintuitive order: from least significant bit (LSB) to most significant bit (MSB).

The top left cell in the viewport is represented by the "first" bit (LSB in this library, not MSB!), and the bottom right cell of the viewport is represented by the "last" bit (MSB).

Now if you think about the left shift as a shift towards the MSB and if you think about the right shift as a shift towards the LSB, it should make more sense.